spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/secretmanager/GcpSecretManagerBootstrapConfiguration.java

```diff
@@ -0,0 +1,111 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.secretmanager;
+
+import java.io.IOException;
+
+import com.google.api.gax.core.CredentialsProvider;
+import com.google.cloud.secretmanager.v1beta1.SecretManagerServiceClient;
+import com.google.cloud.secretmanager.v1beta1.SecretManagerServiceSettings;
+import com.google.protobuf.ByteString;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.bootstrap.config.PropertySourceLocator;
+import org.springframework.cloud.gcp.core.DefaultCredentialsProvider;
+import org.springframework.cloud.gcp.core.DefaultGcpProjectIdProvider;
+import org.springframework.cloud.gcp.core.GcpProjectIdProvider;
+import org.springframework.cloud.gcp.core.UserAgentHeaderProvider;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.core.env.ConfigurableEnvironment;
+
+/**
+ * Bootstrap Autoconfiguration for GCP Secret Manager which enables loading secrets as
+ * properties into the application {@link org.springframework.core.env.Environment}.
+ *
+ * @author Daniel Zou
+ * @since 1.3
+ */
+@Configuration
+@EnableConfigurationProperties(GcpSecretManagerProperties.class)
+@ConditionalOnClass(SecretManagerServiceClient.class)
+@ConditionalOnProperty(value = "spring.cloud.gcp.secretmanager.enabled", matchIfMissing = true)
+public class GcpSecretManagerBootstrapConfiguration {
+
+	private final GcpSecretManagerProperties properties;
+
+	private final CredentialsProvider credentialsProvider;
+
+	private final GcpProjectIdProvider gcpProjectIdProvider;
+
+	public GcpSecretManagerBootstrapConfiguration(GcpSecretManagerProperties properties)
+			throws IOException {
+
+		this.properties = properties;
+		this.credentialsProvider = new DefaultCredentialsProvider(properties);
+		this.gcpProjectIdProvider = properties.getProjectId() != null
+				? properties::getProjectId
+				: new DefaultGcpProjectIdProvider();
+	}
+
+	/**
+	 * Registers {@link ByteString} type converters to convert to String and byte[].
+	 */
+	@Bean
+	public ConfigurableEnvironment configurableEnvironment(
```

Comment:
I wouldn't redefine a `ConfigurableEnvironment` bean. Just put it as a constuctor arg and call a method.

Reply:
Done.